src/main/java/org/folio/inventory/instanceingress/handler/CreateInstanceIngressEventHandler.java

```diff
@@ -0,0 +1,240 @@
+  }
+
+  @Override
+  public CompletableFuture<InstanceIngressEvent> handle(InstanceIngressEvent event) {
+    try {
+      LOGGER.info("Processing InstanceIngressEvent with id '{}' for instance creation", event.getId());
+      if (isNull(event.getEventPayload()) || isNull(event.getEventPayload().getSourceRecordObject())) {
+        var message = format("InstanceIngressEvent message does not contain required data to create Instance for eventId: '%s'",
+          event.getId());
+        LOGGER.error(message);
+        return CompletableFuture.failedFuture(new EventProcessingException(message));
+      }
+      super.getMappingMetadataCache().getByRecordType(InstanceIngressEventConsumer.class.getSimpleName(), context, MARC_BIB_RECORD_TYPE)
+        .map(metadataOptional -> metadataOptional.orElseThrow(() -> new EventProcessingException("MappingMetadata was not found for marc-bib record type")))
+        .compose(mappingMetadataDto -> prepareAndExecuteMapping(mappingMetadataDto, event))
+        .compose(instance -> validateInstance(instance, event))
+        .compose(instance -> saveInstance(instance, event));
+
+      return CompletableFuture.completedFuture(event);
+    } catch (Exception e) {
+      LOGGER.error("Failed to process InstanceIngressEvent with id {}", event.getId(), e);
+      return CompletableFuture.failedFuture(e);
+    }
+  }
+
+  private Future<org.folio.Instance> prepareAndExecuteMapping(MappingMetadataDto mappingMetadata, InstanceIngressEvent event) {
+    return postSnapshotInSrsAndHandleResponse(event.getId())
+      .compose(snapshot -> {
+        try {
+          LOGGER.info("Constructing a Record from InstanceIngressEvent with id '{}'", event.getId());
+          var marcBibRecord = new org.folio.rest.jaxrs.model.Record()
+            .withId(event.getId())
+            .withRecordType(MARC_BIB)
+            .withSnapshotId(event.getId())
+            .withRawRecord(new RawRecord()
+              .withId(event.getId())
+              .withContent(event.getEventPayload().getSourceRecordObject())
+            )
+            .withParsedRecord(new ParsedRecord()
+              .withId(event.getId())
+              .withContent(event.getEventPayload().getSourceRecordObject())
+            );
+          event.getEventPayload()
+            .withAdditionalProperty(MARC_BIBLIOGRAPHIC.value(), marcBibRecord);
+
+          LOGGER.info("Manipulating fields of a Record from InstanceIngressEvent with id '{}'", event.getId());
+          var mappingParameters = Json.decodeValue(mappingMetadata.getMappingParams(), MappingParameters.class);
+          AdditionalFieldsUtil.updateLatestTransactionDate(marcBibRecord, mappingParameters);
+          AdditionalFieldsUtil.move001To035(marcBibRecord);
+          AdditionalFieldsUtil.normalize035(marcBibRecord);
+          AdditionalFieldsUtil.addFieldToMarcRecord(marcBibRecord, TAG_035, SUBFIELD_A, event.getEventPayload().getSourceRecordIdentifier() + BIBFRAME);
```

Comment:
Here we have to prepend the string BIBFRAME
```AdditionalFieldsUtil.addFieldToMarcRecord(marcBibRecord, TAG_035, SUBFIELD_A, BIBFRAME + event.getEventPayload().getSourceRecordIdentifier());```

Reply:
correct, fixed, thank you!